modules/serverpod_auth/serverpod_auth_server/lib/src/endpoints/google_endpoint.dart

```diff
@@ -99,6 +106,8 @@ class GoogleEndpoint extends Endpoint {
       }
     }
 
+    await AuthConfig.current.onUserCreated?.call(session, userInfo);
```

Comment:
Suggestion: Since this depends the operations happening on line 79-107 it would be good to couple them more tightly.

I suggest we implement an `onUserCreatedOverload` callback that replaces the default call and place all of that connected logic inside of that callback instead. Then the last line of that callback would call this row.

Reply:
Yes agreed this is a better setup! ill fix.